tools/mc-java-protoc/src/test/java/io/spine/tools/mc/java/protoc/given/CodeGeneratorRequestGiven.java

```diff
@@ -78,63 +82,83 @@ private CodeGeneratorRequestGiven() {
     }
 
     /**
-     * Creates a {@link SpineProtocConfig} out of the supplied {@code GeneratedMethods} and
-     * a default instance of {@code GeneratedInterfaces} and stores it in the supplied {@code path}.
-     *
-     * @return base64 encoded path to the plugin configuration
-     * @see #protocConfig(Interfaces, Methods, File)
+     * Creates a builder for {@code SpineProtocConfig} with all the validation features turned off.
      */
-    public static String protocConfig(Methods methods, Path configPath) {
-        return protocConfig(new Interfaces(),
-                            methods,
-                            new NestedClasses(),
-                            configPath);
+    public static SpineProtocConfig.Builder configWithoutValidation() {
+        SpineProtocConfig.Builder builder = SpineProtocConfig.newBuilder();
+        builder.getValidationBuilder()
+               .setSkipBuilders(true)
+               .setSkipValidation(true);
+        return builder;
+    }
+
+    /**
+     * Creates a {@link GenerateNestedClasses} config with the given factory class.
+     */
+    public static GenerateNestedClasses generateNested(Class<?> cls) {
+        return GenerateNestedClasses
+                .newBuilder()
+                .setFactory(nestedClassFactory(cls))
+                .build();
     }
 
-    public static String protocConfig(NestedClasses nestedClasses, Path configPath) {
-        return protocConfig(new Interfaces(),
-                            new Methods(),
-                            nestedClasses,
-                            configPath);
+    private static NestedClassFactoryName nestedClassFactory(Class<?> cls) {
+        return NestedClassFactoryName
+                .newBuilder()
+                .setClassName(className(cls))
+                .build();
     }
 
     /**
-     * Creates a {@link SpineProtocConfig} out of the supplied {@code GeneratedInterfaces} and
-     * a default instance of {@code GeneratedMethods} and stores it in the supplied {@code path}.
-     *
-     * @return base64 encoded path to the plugin configuration
-     * @see #protocConfig(Interfaces, Methods, File)
+     * Creates a {@link GenerateMethods} config with the given factory class.
+     */
+    public static GenerateMethods generateMethods(Class<?> factory) {
+        MethodFactoryName factoryName = methodFactory(factory);
+        return GenerateMethods
+                .newBuilder()
+                .setFactory(factoryName)
+                .build();
+    }
+
+    public static MethodFactoryName methodFactory(Class<?> cls) {
```

Comment:
As long as we started documenting `public static` API level in this type, I'd propose to do it here as well.